File: ccbusDemo/server/ccbusdemo/src/main/java/ccbus/demo/ccbus/service/inner/ServiceSome.java
package ccbus.demo.ccbus.service.inner;

import ccbus.connect.core.ccbus.WorkerService;
import ccbus.demo.ccbus.payload.SomePayload;


public class ServiceSome extends WorkerService<SomePayload>
{
    public String someVal="Hello";
    private int ival=1;

    @Override
    public void compute()
    {
        int a=2;
        int b=3;
        ival=2+3;
        String val=someVal;
        this.complete(new SomePayload(1,"name"));
    }

    public void setSomeVal(String someVal)
    {
        this.someVal = someVal;
    }
}

File: ccbusDemo/client/ccbusdemo/src/service/ServiceSome.js
import {WorkerService} from 'ccbus/WorkerService';
import {FileDownload} from 'ccbus/payload/FileDownload';
import {SomePayload} from 'payload/SomePayload';
import {WorkerServer} from 'ccbus';


/**
* @extends {WorkerService<SomePayload>}
*/ 
export class ServiceSome extends WorkerService
{
    
    /**
    * @type {String}
    */ 
    someVal = "Hello";
    
    /**
    * @type {number}
    */ 
    ival = 1;
    
    /**
    * @param {String} someVal
    * @returns {void}
    */ 
    setSomeVal(someVal)
    {
        this.someVal=someVal;
    }
    
    /**
    * @returns {}
    */ 
    compute()
    {
        WorkerServer.computeService({
            
            /**
            * @type {String}
            */ 
            endpoint: "service.ServiceSome",
            
            /**
            * @type {Object}
            */ 
            data: {
                
                /**
                * @type {number}
                */ 
                ival: this.ival,
                
                /**
                * @type {String}
                */ 
                someVal: this.someVal
            },
            
            /**
            * @type {Object}
            */ 
            bind: this,
            
            /**
            * @type {Object}
            */ 
            args: arguments
        })
    }
}